src/App.tsx

```diff
@@ -1,24 +1,62 @@
-import { Box } from "@mui/material";
-import RootNoteSelect from "./components/SelectRootNote";
+import { Box, Typography } from '@mui/material';
+import { useState } from 'react';
+import RootNoteSelect from './components/SelectRootNote';
 import TuningSelector from './components/TuningSelector';
-import ModeSelector from "./components/ModeSelector";
-import KeyAndTuningButton from "./components/KeyAndTuningButton"
-import CompensateForTuningOption from "./components/CompensateForTuningOption"
-import Typography from '@mui/material/Typography';
-
-export default function App() {
-  return <Box sx={{ display: 'flex', flexDirection: 'column', gap: 2 }}>
-    <RootNoteSelect />
-    <ModeSelector />
-    <TuningSelector />
-    <Typography variant="body1" sx={{ mt: 2 }}>
-      <strong>Compensate note positions for different tunings?</strong>{' '}
-      <br />
-      If <strong>"Yes"</strong>: The low C in C Standard will be referred to as strictly C, and C minor will have the same fingering and position as E minor in E Standard tuning.
-      <br />
-      If <strong>"No"</strong>: The low C in C Standard will be referred to as E, and E minor will retain the same fingering and note position as it would in E Standard.
-    </Typography>
-    <CompensateForTuningOption />
-    <KeyAndTuningButton />
-  </Box>
-}
\ No newline at end of file
+import ModeSelector from './components/ModeSelector';
+import KeyAndTuningButton from './components/KeyAndTuningButton';
+import CompensateForTuningOption from './components/CompensateForTuningOption';
+
+interface RootOption {
+  value: string;
+  label: string;
+}
+
+const App = () => {
+  const [selectedRoot, setSelectedRoot] = useState<RootOption | null>(null);
+  const [selectedMode, setSelectedMode] = useState<string | null>(null);
+  const [selectedTuning, setSelectedTuning] = useState<string | null>(null);
+  const [stringTunings, setStringTunings] = useState<string[]>(Array(6).fill(""));
+
+  const handleRootSelect = (root: RootOption | null) => {
+    setSelectedRoot(root);
+  };
+
+  const handleModeSelect = (mode: string | null) => {
+    setSelectedMode(mode);
+  };
+
+  const handleTuningSelect = (tuning: string | null, stringTunings: string[]) => {
+    setSelectedTuning(tuning);
+    setStringTunings(stringTunings);
+  };
+
+  const isIncomplete =
+    !selectedRoot ||
+    !selectedMode ||
+    !selectedTuning ||
+    (selectedTuning === 'Other' && stringTunings.some((tuning) => !tuning));
+
+  return (
+    <Box sx={{ display: 'flex', flexDirection: 'column', gap: 2 }}>
+      <RootNoteSelect onSelect={handleRootSelect} />
+      <ModeSelector onSelect={handleModeSelect} />
+      <TuningSelector onSelect={handleTuningSelect} />
```

Comment:
@stenzeladam It is best to place all things related to the Tuning (Typography, Switch) in a separate specific component.

Reply:
Done, moved the Typography and Switch into "CompensateForTuningOption.tsx".